src/cavendish_particle_tracks/_widget.py

```diff
@@ -180,26 +212,75 @@ def _get_table_column_index(self, columntext: str) -> int:
 
     def _on_row_selection_changed(self) -> None:
         """Enable/disable calculation buttons depending on the row selection"""
+        self.set_btn_availability()
+
+    def set_btn_availability(self) -> None:
+        images_imported = False
+        for layer in self.viewer.layers:
+            if layer.name == "Particle Tracks":
+                images_imported = True
+                break
+        self.set_UI_image_loaded(images_imported, self.test_mode)
         try:
             selected_row = self._get_selected_row()
+            self.btn_save.setEnabled(True)
+            self.btn_delete_particle.setEnabled(True)
+            ## think about these two + cal once done.
+            self.btn_magnification.setEnabled(True)
+            self.btn_stereoshift.setEnabled(True)
             if self.data[selected_row].index < 4:
-                self.rad.setEnabled(True)
-                self.lgth.setEnabled(True)
-                self.ang.setEnabled(False)
+                self.btn_radius.setEnabled(True)
+                self.btn_length.setEnabled(True)
+                self.btn_decayangle.setEnabled(False)
                 return
             elif self.data[selected_row].index == 4:
-                self.rad.setEnabled(False)
-                self.lgth.setEnabled(True)
-                self.ang.setEnabled(True)
+                self.btn_radius.setEnabled(False)
+                self.btn_length.setEnabled(True)
+                self.btn_decayangle.setEnabled(True)
                 return
         except IndexError:
-            print("The table is empty.")
-        self.disable_all_buttons()
-
-    def disable_all_buttons(self) -> None:
-        self.rad.setEnabled(False)
-        self.lgth.setEnabled(False)
-        self.ang.setEnabled(False)
+            self.btn_delete_particle.setEnabled(False)
+            self.btn_radius.setEnabled(False)
+            self.btn_length.setEnabled(False)
+            self.btn_decayangle.setEnabled(False)
+            self.cal.setEnabled(False)
+            self.btn_stereoshift.setEnabled(False)
+            self.btn_magnification.setEnabled(False)
+            self.btn_save.setEnabled(False)
+
+    def set_UI_image_loaded(self, loaded: bool, test_mode: bool) -> None:
+        if test_mode:
+            return
```

Comment:
test mode is still appropriate here, since what you're mentioning is the next else statement down.

Will get a docstring in here to make it clearer but the decision its making is:

if test mode -> leave all UI elements visible
else -> if (data loaded) -> show UI elements except for those from the splash screen.
        -> else if (data not loaded) -> block widget without loaded data.

hence, the blocking of UI elements only occurs if the application isn't in test mode.